pwiz_tools/Skyline/Model/Proteome/ProteinAssociation.cs

```diff
@@ -561,11 +610,65 @@ public void ApplyParsimonyOptions(bool groupProteins, bool findMinimalProteinLis
             _finalResults.FinalSharedPeptideCount = sharedPeptidesRemaining.Values.Sum();
         }
 
-        private Dictionary<IProteinRecord, PeptideAssociationGroup> CalculateProteinGroups(MappingResultsInternal results, ILongWaitBroker broker)
+        public class GeneLevelEqualityComparer : EqualityComparer<IProteinRecord>

```

Comment:
Instead of writing an EqualityComparer, I probably would have written a new class called "GeneOrProteinSequence" which I used as the Key in the dictionary.
The reason is that it's easier to understand what is going on if the dictionary has a completely different key type, compared to if it has a special EqualityComparer.

But, this way is fine, too.

Reply:
After I wrote this, I realized I could have just done a string compare with the gene name (or protein name if gene isn't set). Does that seem sensible to you as well?